src/halmos/sevm.py

```diff
@@ -702,6 +757,15 @@ def halt(
     def is_halted(self) -> bool:
         return self.context.output.data is not None
 
+    def reverted_with(self, expected: ByteVec) -> bool:
+        if not isinstance(self.context.output.error, Revert):
+            return False
+
+        returndata = self.context.output.data[: byte_length(expected)]
+
+        # bytevec equality check, will take care of length check, bv vs symbolic, etc.
+        return returndata == expected
+
```

Comment:
nice! but why do we need to compare only the prefix of data, not the full data?

Reply:
equality check fails if length is different, do you think that's closer to what we want?